sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -1314,99 +1312,6 @@ class SumBinaryFixer(toType: DataType, isAnsi: Boolean)
   }
 }
 
-/**
- * Fixes up a sum operation for unbounded preceding to unbounded following
- * @param errorOnOverflow if we need to throw an exception when an overflow happens or not.
- */
-class SumUnboundedToUnboundedFixer(resultType: DataType, failOnError: Boolean)
-    extends BatchedUnboundedToUnboundedWindowFixer {
-
-  private var previousValue: Option[Scalar] = None
-
-  private val numeric = TypeUtils.getNumeric(resultType, failOnError)
-
-  override def updateState(scalar: Scalar): Unit = {
-    if (scalar.isValid) {
-      previousValue match {
-        case None =>
-          previousValue = Some(scalar.incRefCount())
-        case Some(prev) =>
-          withResource(prev) { _ =>
-            previousValue = None
-            prev.getType.getTypeId match {
-              case DType.DTypeEnum.INT64 =>
-                if (failOnError) {
-                  previousValue = Some(Scalar.fromLong(Math.addExact(
-                    scalar.getLong, prev.getLong)))
-                } else {
-                  previousValue = Some(Scalar.fromLong(scalar.getLong + prev.getLong))
-                }
-              case DType.DTypeEnum.FLOAT32 =>
-                previousValue = Some(Scalar.fromFloat(scalar.getFloat + prev.getFloat))
-              case DType.DTypeEnum.FLOAT64 =>
-                previousValue = Some(Scalar.fromDouble(scalar.getDouble + prev.getDouble))
-              case DType.DTypeEnum.DECIMAL32 | DType.DTypeEnum.DECIMAL64 |
-                   DType.DTypeEnum.DECIMAL128 =>
-                val decimal = numeric.plus(Decimal(prev.getBigDecimal),
-                  Decimal(scalar.getBigDecimal)).asInstanceOf[Decimal]
-                val dt = resultType.asInstanceOf[DecimalType]
-                previousValue = Option(TrampolineUtil.checkDecimalOverflow(
-                    decimal, dt.precision, dt.scale, failOnError))
-                  .map(n => Scalar.fromDecimal(n.toJavaBigDecimal))
-              case other =>
-                throw new IllegalStateException(s"unhandled type: $other")
-            }
-          }
-        }
-    }
-  }
-
-  override def fixUp(
-      samePartitionMask: Either[ColumnVector, Boolean],
-      column: ColumnVector): ColumnVector = {
-
-    def makeScalar(): Scalar = previousValue match {
-      case Some(value) =>
-        value
-      case _ => resultType match {
-        case DataTypes.LongType => Scalar.fromNull(DType.INT64)
-        case DataTypes.FloatType => Scalar.fromNull(DType.FLOAT32)
-        case DataTypes.DoubleType => Scalar.fromNull(DType.FLOAT64)
-        case d: DecimalType =>
-          val dt = if (d.precision > DType.DECIMAL64_MAX_PRECISION) {
-            DType.DTypeEnum.DECIMAL128.getNativeId
-          } else if (d.precision > DType.DECIMAL32_MAX_PRECISION) {
-            DType.DTypeEnum.DECIMAL64.getNativeId
-          } else {
-            DType.DTypeEnum.DECIMAL32.getNativeId
-          }
-          Scalar.fromNull(DType.fromNative(dt, d.scale))
```

Comment:
Just FYI I think this is the problem here. d.scale is a Spark scale, but DType wants `-d.scale`